packages/augur-core/source/contracts/reporting/ShareToken.sol

```diff
@@ -242,6 +241,7 @@ contract ShareToken is ITyped, Initializable, ERC1155, IShareToken, ReentrancyGu
         require(isApprovedForAll(_longParticipant, msg.sender) == true, "ERC1155: need operator approval to burn long account shares");
 
         _internalTransferFrom(_shortParticipant, _longParticipant, getTokenId(_market, _outcome), _amount, bytes(""), false);
+        // NOTE: burnCompleteSets will validate the market provided is legitimate
         (uint256 _payout, uint256 _creatorFee, uint256 _reportingFee) = burnCompleteSets(_market, _longParticipant, _amount, _sourceAccount, _fingerprint);
 
         uint256 _longPayout = _payout.mul(_price) / _market.getNumTicks();
```

Comment:
what do you think of using 

`markets[address(_market)].numTicks` instead of `_market.getNumTicks()`

Reply:
Causes stack depth issues sadly

Reply:
That's too bad! it's definitely an unusual way to get numTicks when you are guaranteed to have it locally/cheaper/safer! Worth a comment?

And ya, burnSets will def protect from the untrusted market.